modules/client-handler/src/main/java/org/apache/ignite/client/handler/ClientInboundMessageHandler.java

```diff
@@ -661,49 +661,49 @@ private void processOperation(ChannelHandlerContext ctx, ClientMessageUnpacker i
                 return ClientTableGetRequest.process(in, out, igniteTables);
 
             case ClientOp.TUPLE_UPSERT:
-                return ClientTupleUpsertRequest.process(in, out, igniteTables, resources);
+                return ClientTupleUpsertRequest.process(in, out, igniteTables, resources, igniteTransactions);
```

Comment:
We don't need to track a client observable ts on server because it's already tracked on client for RO purposes:
`org.apache.ignite.internal.client.ReliableChannel#observableTimestamp`
This should be fixed. Instead, the commit ts should be propagated directly to client on commit.

Reply:
Currently, we can't do that. Probably we might think about this later.